test/common/io/io_uring_worker_impl_integration_test.cc

```diff
@@ -0,0 +1,270 @@
+    IoUringSocketEntry::onConnect(req, result, injected);
+    connect_result_ = result;
+    is_connect_injected_completion_ = injected;
+    nr_completion_++;
+  }
+  void onRead(Request* req, int32_t result, bool injected) override {
+    IoUringSocketEntry::onRead(req, result, injected);
+    read_result_ = result;
+    is_read_injected_completion_ = injected;
+    nr_completion_++;
+  }
+  void onWrite(Request* req, int32_t result, bool injected) override {
+    IoUringSocketEntry::onWrite(req, result, injected);
+    write_result_ = result;
+    is_write_injected_completion_ = injected;
+    nr_completion_++;
+  }
+  void onClose(Request* req, int32_t result, bool injected) override {
+    IoUringSocketEntry::onClose(req, result, injected);
+    close_result_ = result;
+    is_close_injected_completion_ = injected;
+    nr_completion_++;
+  }
+  void onCancel(Request* req, int32_t result, bool injected) override {
+    IoUringSocketEntry::onCancel(req, result, injected);
+    cancel_result_ = result;
+    is_cancel_injected_completion_ = injected;
+    nr_completion_++;
+  }
+  void onShutdown(Request* req, int32_t result, bool injected) override {
+    IoUringSocketEntry::onShutdown(req, result, injected);
+    shutdown_result_ = result;
+    is_shutdown_injected_completion_ = injected;
+    nr_completion_++;
+  }
+
+  void cleanupForTest() { cleanup(); }
+
+  int32_t accept_result_{-1};
+  bool is_accept_injected_completion_{false};
+  int32_t connect_result_{-1};
+  bool is_connect_injected_completion_{false};
+  int32_t read_result_{-1};
+  bool is_read_injected_completion_{false};
+  int32_t write_result_{-1};
+  bool is_write_injected_completion_{false};
+  int32_t nr_completion_{0};
+  int32_t close_result_{-1};
+  bool is_close_injected_completion_{false};
+  int32_t cancel_result_{-1};
+  bool is_cancel_injected_completion_{false};
+  int32_t shutdown_result_{-1};
+  bool is_shutdown_injected_completion_{false};
+};
+
+class IoUringWorkerTestImpl : public IoUringWorkerImpl {
+public:
+  IoUringWorkerTestImpl(IoUringPtr io_uring_instance, Event::Dispatcher& dispatcher)
+      : IoUringWorkerImpl(std::move(io_uring_instance), dispatcher) {}
+
+  IoUringSocket& addTestSocket(os_fd_t fd) {
+    return addSocket(std::make_unique<IoUringSocketTestImpl>(fd, *this));
+  }
+
+  const std::list<IoUringSocketEntryPtr>& getSockets() const { return sockets_; }
+};
+
+class IoUringWorkerIntegrationTest : public testing::Test {
+public:
+  IoUringWorkerIntegrationTest() : should_skip_(!isIoUringSupported()) {}
+
+  void SetUp() override {
+    if (should_skip_) {
+      GTEST_SKIP();
+    }
+  }
+
+  void initialize() {
+    api_ = Api::createApiForTest(time_system_);
+    dispatcher_ = api_->allocateDispatcher("test_thread");
+    io_uring_worker_ = std::make_unique<IoUringWorkerTestImpl>(
+        std::make_unique<IoUringImpl>(20, false), *dispatcher_);
+  }
+  void initializeSockets() {
+    socket(true, true);
```

Comment:
could this be named something more clear such as `createServerListenerAndClientSocket` , it's a bit hard to read otherwise

Reply:
got it, will change it